rest/method_get.go

```diff
@@ -43,7 +43,7 @@ func listGet(ctx context.Context, r *http.Request, route *RouteMatch) (status in
 		list.Offset = win.Offset
 	}
 	for _, item := range list.Items {
-		item.Payload, err = q.Projection.Eval(ctx, item.Payload, rsc.Validator(), getReferenceResolver(rsc))
+		item.Payload, err = q.Projection.Eval(ctx, item.Payload, restResource{rsc})
```

Comment:
Just trying to compare this against the `query.Selection` idea, not trying to state which one is better:

With the `Selection` idea, this would be (without the for-loop on line 45):

```go
payloads, err = q.Projection.Eval(ctx, restSelection{resource: rsc, items: list.Items})
if err != nil {...}
for i := payloads {
    list.Items[i] = payloads[i]
}
```